crypto/evp_extra/p_dh_asn1.c

```diff
@@ -12,10 +12,71 @@
 #include <openssl/bn.h>
 #include <openssl/dh.h>
 #include <openssl/err.h>
+#include <openssl/x509.h>
 
-#include "internal.h"
-#include "../internal.h"
 #include "../fipsmodule/cpucap/internal.h"
+#include "../fipsmodule/dh/internal.h"
+#include "../internal.h"
+#include "internal.h"
+
+static int dh_pub_encode(CBB *out, const EVP_PKEY *key) {
```

Comment:
should we null-check `key` before proceeding?

Reply:
That's a good point.  The caller [does not check for null](https://github.com/aws/aws-lc/blob/4916fe8641c5c220e6e6921defc026a8c564f855/crypto/evp_extra/evp_asn1.c#L144-L151) and neither does the [equivalent RSA function](https://github.com/aws/aws-lc/blob/4916fe8641c5c220e6e6921defc026a8c564f855/crypto/evp_extra/p_rsa_asn1.c#L69-L83). I think there should be a NULL check at the top-level.